### PR TITLE
GU UP Migration Pre-Work

### PR DESCRIPTION
**Pull request by oslynn**
This PR GU UP Migration Pre-Work

Test Steps:
1. *Include steps to test these changes*

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue
https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/14881

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/settings/STLTs/GU/gu-doh.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  featureFlags: null
+  keys: null
+  senders: []
+  receivers:
+    - name: "full-elr"
+      organizationName: "gu-doh"
+      topic: "full-elr"
+      customerStatus: "active"
+      translation:
+        schemaName: "classpath:/metadata/hl7_mapping/receivers/STLTs/GU/GU-receiver-transform.yml"
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/GU/gu-doh.yml:17]: Needs to be azure without the metadata

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/GU/GU-receiver-transform.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    hl7Spec: [ 'MSH-3-3' ]
+
+  - name: gu-name-of-the-receiving-application
+    condition: 'true'
+    value: [ '"GUDOH"' ]
+    hl7Spec: [ 'MSH-5-1' ]
+
+  - name: gu-receiving-facility-namespace-ID
+    condition: 'true'
+    value: [ '"FDOH"' ]
```
Comment (JessicaWNava) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/GU/GU-receiver-transform.yml:39]: I think this needs to be GUDOH, that is what was in their replaceAwithB setting
"MSH-6": [
      {
        "*": "GUDOH"
      }
    ]

File: prime-router/settings/STLTs/GU/gu-doh.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        - "(Bundle.entry.resource.ofType(Patient).address.line.exists() or Bundle.entry.resource.ofType(Patient).address.postalCode.exists()\
+      \ or Bundle.entry.resource.ofType(Patient).telecom.exists())"
+        - "Bundle.entry.resource.ofType(MessageHeader).id.exists()"
+        - "Bundle.entry.resource.ofType(Patient).birthDate.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.family.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.given.count() > 0"
+        - "Bundle.entry.resource.ofType(Specimen).type.exists()"
+      routingFilter:
+        # - "matches(test_result_status, F, C, P)"
+        - "Bundle.entry.resource.ofType(DiagnosticReport).where(status in 'final'|'corrected'|'preliminary'|'amended').exists()"
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/GU/gu-doh.yml:75]: I believe @victor-chaparro said these need to go in the processingModeFilter

File: prime-router/settings/STLTs/GU/gu-doh.yml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      reverseTheQualityFilter: false
+      conditionFilter:
+        # Accept COVID only
+        - "(%resource.code.coding.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code').value.where(code  in ('840539006')).exists())"
+      mappedConditionFilter: []
+      deidentify: false
+      deidentifiedValue: ""
+      timing:
+        operation: "MERGE"
+        numberPerDay: 1440
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/GU/gu-doh.yml:86]: Needs to be adjusted back to 12 like they had

File: prime-router/settings/STLTs/GU/gu-doh.yml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        useHighPrecisionHeaderDateTimeFormat: false
+        type: "HL7"
+        truncationConfig:
+          truncateHDNamespaceIds: false
+          truncateHl7Fields: []
+          customLengthHl7Fields: {}
+      jurisdictionalFilter:
+        - "(Bundle.entry.resource.ofType(ServiceRequest)[0].requester.resolve().organization.resolve().address.state = 'GU') or (Bundle.entry.resource.ofType(Patient).address.state = 'GU')"
+      qualityFilter:
+        - "((Bundle.entry.resource.ofType(Specimen).collection.collectedPeriod.exists()\
```
Comment (JessicaWNava) [inline prime-router/settings/STLTs/GU/gu-doh.yml:63]: This looks different from the default quality filter and I didn't see anything in their settings to indicate they should have anything other than that.

**Review by JessicaWNava — CHANGES_REQUESTED** [review]
Looks good, but some changes needed.

File: prime-router/settings/STLTs/GU/gu-doh.yml (hunk 1 above)
Comment (oslynn) [reply prime-router/settings/STLTs/GU/gu-doh.yml:17]: Thank you very much.

File: prime-router/settings/STLTs/GU/gu-doh.yml (hunk 3 above)
Comment (oslynn) [reply prime-router/settings/STLTs/GU/gu-doh.yml:75]: Ok, thanks

File: prime-router/settings/STLTs/GU/gu-doh.yml (hunk 4 above)
Comment (oslynn) [reply prime-router/settings/STLTs/GU/gu-doh.yml:86]: Thanks

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/GU/GU-receiver-transform.yml (hunk 2 above)
Comment (oslynn) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/GU/GU-receiver-transform.yml:39]: Good catched.

File: prime-router/settings/STLTs/GU/gu-doh.yml (hunk 5 above)
Comment (oslynn) [reply prime-router/settings/STLTs/GU/gu-doh.yml:63]: @JessicaWNava, I am not sure that I understand this comment.  What is the default quality filter that you referring to?

**Review by JessicaWNava — APPROVED** [review]
LGTM!

**Pull request merged**